server/internal/usecase/interactor/integration.go

```diff
@@ -130,6 +131,20 @@ func (i Integration) Delete(ctx context.Context, integrationId id.IntegrationID,
 			if in.Developer() != *operator.User {
 				return interfaces.ErrOperationDenied
 			}
+
+			// remove the integration from the connected workspaces
+			ws, err := i.repos.Workspace.FindByIntegration(ctx, integrationId)
+			if err != nil && !errors.Is(err, rerror.ErrNotFound) {
+				return err
+			}
+			for _, w := range ws {
+				if err := w.Members().DeleteIntegration(integrationId); err != nil {
+					return err
+				}
+				if err := i.repos.Workspace.Save(ctx, w); err != nil {
```

Comment:
The current implementation is acceptable, but if possible, please implement SaveAll in mongo.Workspace to save all workspaces at once. The fewer interactions with the database, the better.